src/industry_gui.cpp

```diff
@@ -578,9 +589,53 @@ class BuildIndustryWindow : public Window {
 		}
 	}
 
+	static void AskManyRandomIndustriesCallback(Window *w, bool confirmed)
+	{
+		if (!confirmed) return;
+
+		if (Town::GetNumItems() == 0) {
+			ShowErrorMessage(STR_ERROR_CAN_T_GENERATE_INDUSTRIES, STR_ERROR_MUST_FOUND_TOWN_FIRST, WL_INFO);
+		}
+		else {
+			extern void GenerateIndustries();
+			_generating_world = true;
+			GenerateIndustries();
+			_generating_world = false;
+		}
+	}
+
+	static void AskRemoveAllIndustriesCallback(Window *w, bool confirmed)
+	{
+		if (!confirmed) return;
+
+		for (Industry* industry : Industry::Iterate()) delete industry;
+
+		for (TileIndex tile = 0; tile < MapSize(); tile++) {
+			if (IsTileType(tile, MP_CLEAR) && GetRawClearGround(tile) == CLEAR_FIELDS) {
+				MakeClear(tile, CLEAR_GRASS, 3); // Clear farmland
```

Comment:
Please put this comment before the `for`. The whole `for` is about clearing farmland, not only this `MakeClear` :)